### PR TITLE
Add team API client with graceful degradation

### LINKED ISSUE #6: Team API client in MCP server
## Description

Add an HTTP client to the MCP server that queries and writes to the team API. This connects the two tiers: local store + team store.

Ref: [design doc — Architecture](docs/plans/2026-03-04-craic-mvp-poc-design.md#architecture)

## Deliverables

- `server/craic_mcp/team_client.py` — async HTTP client for team API
- Team API URL configurable via environment variable (default `http://localhost:8742`)
- Update `craic_query` to search local → team, merge and dedup results
- Update `craic_propose` to push generic insights to team store
- Update `craic_confirm` and `craic_flag` to propagate to team store if unit exists there
- Graceful degradation if team API is unreachable (local-only mode)
- Integration tests (with team API running)

## Depends on

- #4 (MCP server)
- #5 (team API)

### PR DESCRIPTION
**Pull request by peteski22**
## Summary

- Add `TeamClient` HTTP wrapper for all team API endpoints with full graceful degradation (transport errors, malformed JSON, schema mismatches all return `None`).
- Integrate team client into MCP server: `craic_query` merges local+team results with dedup and re-ranking; `craic_propose` pushes to team best-effort; `craic_confirm`/`craic_flag` propagate to team.
- Thread-safe lifecycle management: store registry closes all thread-local SQLite connections at shutdown; double-checked locking prevents duplicate team client creation.
- Source indicator reflects which stores were *consulted*, not just which tiers survived deduplication.

## Changes

- **`server/craic_mcp/team_client.py`** (new) — synchronous HTTP client with context manager support
- **`server/craic_mcp/server.py`** — integrated team client with merge/dedup logic, store registry, degradation logging
- **`server/tests/test_team_client.py`** (new) — 14 tests covering success, error, and edge cases
- **`server/tests/test_server.py`** — 45 tests including team integration scenarios

## Test plan

- [x] All 166 tests pass (`make test`)
- [x] Lint clean (`make lint`)
- [x] Type check clean (`ty check`)
- [x] Star-chamber 2-round debate review (3 providers) — all consensus items addressed

Closes #6

### CODE REVIEW

File: server/craic_mcp/server.py [hunk 1]
```diff
@@ -15,43 +19,162 @@
...
-    if store is not None:
-        store.close()
+    """Close all registered stores across all threads."""
+    with _store_registry_lock:
+        for store in _store_registry:
+            store.close()
+        _store_registry.clear()
+    # Also clear the thread-local reference for the current thread.
+    if getattr(_store_local, "store", None) is not None:
         _store_local.store = None
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:78]: Fixed — docstring now documents the shutdown-only assumption and notes that thread-local references on other threads become stale.

File: server/craic_mcp/server.py [hunk 2]
```diff
@@ -15,43 +19,162 @@
...
+        return _team_client
+    with _team_client_lock:
+        # Double-check after acquiring the lock.
+        if _team_client is not None:
+            return _team_client
+        url = os.environ.get("CRAIC_TEAM_API_URL", _DEFAULT_TEAM_API_URL)
+        if not url:
+            return None
+        _team_client = TeamClient(base_url=url)
+    return _team_client
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:105]: Fixed — added a `_DISABLED_SENTINEL` so the fast path skips the lock when team is disabled.

File: server/craic_mcp/server.py [hunk 3]
```diff
@@ -15,43 +19,162 @@
...
+        team_units: Results from the team API, or None if unavailable.
+        query_domains: Domain tags used in the query.
+        query_language: Language filter used in the query.
+        query_framework: Framework filter used in the query.
+        limit: Maximum results to return.
+
+    Returns:
+        Tuple of (serialised results, source indicator). The source
+        reflects whether each store was *consulted*, not just whether
+        its results survived deduplication.
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed — docstring now says "consulted and returned results" to match the actual logic.

File: server/craic_mcp/server.py [hunk 4]
```diff
@@ -63,15 +186,18 @@ def craic_query(
...
         framework: Optional framework filter.
         limit: Maximum results to return.
 
     Returns:
-        Dict with ``results`` (list of knowledge unit dicts) and ``source``,
-        or ``error`` if inputs are invalid.
+        Dict with ``results`` (list of knowledge unit dicts) and ``source``
+        ("local", "team", or "both"), or ``error`` if inputs are invalid.
     """
     cleaned = [d.strip() for d in domain if d.strip()]
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed — domain tags are now lowercased during cleaning to match the normalisation applied during storage.

File: server/craic_mcp/server.py [hunk 5]
```diff
@@ -114,17 +275,21 @@ def craic_propose(
...
+        Dict with ``id``, ``tier``, ``message``, and ``team_id``
+        (if pushed to team), or ``error`` if inputs are invalid.
     """
-    if not summary.strip() or not detail.strip() or not action.strip():
+    cleaned_summary = summary.strip()
+    cleaned_detail = detail.strip()
+    cleaned_action = action.strip()
+    if not cleaned_summary or not cleaned_detail or not cleaned_action:
         return {"error": "summary, detail, and action must be non-blank."}
     cleaned_domain = [d.strip() for d in domain if d.strip()]
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed — craic_propose now lowercases domain tags to match craic_query.
